guides/provide-protection.ts

```diff
@@ -0,0 +1,84 @@
+  const { parseUnits } = hre.ethers.utils;
+
+  // Since we are testing on a forked Rinkeby and our account has no tokens, we need to initialize the account with
+  // the required tokens
+  await supplyTokensTo('USDC', supplyAmount, signer.address, hre, signer);
+
+  // STEP 1: VERIFY MARKET
+  // We know we'll need the Comptroller, so create an instance the Comptroller contract
+  const comptrollerAddress = getContractAddress('Comptroller', chainId);
+  const comptroller = new Contract(comptrollerAddress, comptrollerAbi, signer); // connect signer for sending transactions
+
+  // The first check is to make sure our protection market is a valid protection market that we can supply to
+  const cUsdcProtectionMarketAddr = '0xbb0e6fC1FE1913DF8f0A483E6dC48329871cb2b7'; // this is the protection market address
+  const allMarkets = await comptroller.getAllMarkets();
+  if (!allMarkets.includes(cUsdcProtectionMarketAddr)) {
+    logFailure("Provided Protection Market address not found in the Comptroller's list of all markets");
+  }
+  logSuccess('Provided Protection Market address is valid');
+
+  // STEP 2: PROVIDE PROTECTION
+  // We're now ready to supply collateral (i.e. provide protection) to the market, but there's some preparation we
+  // need to do beforehand. First, recall that USDC has 6 decimal places, so we need to take that into account. We'll
+  // do this programmatically by querying the USDC contract for the number of decimals it has
+  const usdcAddress = getContractAddress('USDC', chainId); // use our helper method to get the USDC contract address
+  const usdc = new Contract(usdcAddress, erc20Abi, signer);
+  const decimals = await usdc.decimals();
+  const parsedSupplyAmount = parseUnits(supplyAmount, decimals); // scale amount based on number of decimals
+
+  // Next we need to approve the protection market contract to spend our USDC. We trust the contract, so approve it to
+  // spend the maximum possible amount to avoid future approvals and save gas
+  const approveTx = await usdc.approve(cUsdcProtectionMarketAddr, MaxUint256);
+  await approveTx.wait();
+
+  // Let's verify this approve transaction was successful
+  const allowance = await usdc.allowance(signer.address, cUsdcProtectionMarketAddr);
+  if (!allowance.eq(MaxUint256)) {
+    logFailure('CozyUSDC does not have sufficient allowance to spend our USDC. Exiting script');
+    return;
+  }
+  logSuccess('Approval transaction successful. Ready to mint CozyUSDC with our USDC');
+
+  // Now we can supply funds to the protection market to provide protection. Just like with ordinary Money Markets,
+  // this mints a receipt token that is sent to our wallet
+  const cUsdcProtectionMarket = new Contract(cUsdcProtectionMarketAddr, cozyTokenAbi, signer);
+  const mintTx = await cUsdcProtectionMarket.mint(parsedSupplyAmount);
+  const { log: mintLog, receipt: mintReceipt } = await findLog(mintTx, cUsdcProtectionMarket, 'Mint', provider);
```

Comment:
nit1: Since `mintLog` is not used, prefix with `_` to avoid TypeScript eror

nit2: Not strictly part of this PR, but reading this script, it was not obvious that `findLog` throws if the log is not found, and I forgot this from previous reviews. Had to source dive to confirm that was the case. Might consider a more explicit name like `assertExtractLog` or `findLogOrThrow` for easier reading of these scripts.

Reply:
Regarding nit1: Where were you seeing a TS error here? For me the script runs, and yarn lint also doesn’t show an error. It only indicates something if I hover over the variable, but I think that must be from VSCode if compiling doesn't complain. Interestingly it seems adding the underscore also removes the syntax highlighting that makes it clear we are aren’t using that variable 🤔

Regarding nit2: *should* the name convey that it throws? Most ethers.js methods throw when they can't do what you ask them, but aren't named as such, e.g. `getAddress()` and `parseLog()` throw if they can't do what it sounds like they do